### PR TITLE
Fix all the hovers.

### PR DESCRIPTION
**Pull request by kadams54**
Corrects a number of hover-related issues around the app:

* Fixes https://github.com/juju/juju-gui/issues/1254
* Fixes https://github.com/juju/juju-gui/issues/1251
* Fixes https://github.com/juju/juju-gui/issues/1249
* Fixes https://github.com/juju/juju-gui/issues/1242

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4214/
Test PASSed.

**Comment by kadams54** [comment]
## QA

1. For #1254: In Chrome, ensure that the transition is happening on the placeholder text, as well as the surrounding icons (magnifying glass, "Store" text and shopping cart). Note that FF doesn't yet support setting a transition on the placeholder, so it's still a bit janky. A JSFiddle for testing browser support: https://jsfiddle.net/z74xuxjm/
2. For #1251: Create some machines and containers; hovering should change the background to white.
3. For #1249: Deploy a service, then verify that hovering over the close icon in the deployment summary changes the icon to orange.
4. For #1242: Hover over the Import/Export links and verify that underlines appear.

**Comment by hatched** [comment]
For the hovers in the machine view can you add a slight fade in/out

**Comment by hatched** [comment]
:+1: :ok: With one small request
Thanks, these look good!

**Comment by jujugui** [comment]
Merged build finished. Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4215/
Test FAILed.

**Comment by kadams54** [comment]
Not sure what's going on with CI here. A manually-started CI build finished successfully: http://ci.jujugui.org:8080/job/juju-gui/4216/

File: jujugui/static/gui/src/app/components/deployment-bar/_deployment-bar.scss [hunk 1]
```diff
@@ -35,4 +35,8 @@
 %deployment-bar__links {
     position: absolute;
     top: 11px;
+
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-bar/_deployment-bar.scss]: Can you also add
```
&:focus,
&:active,
```

File: jujugui/static/gui/src/app/components/deployment-summary/_deployment-summary.scss [hunk 2]
```diff
@@ -18,6 +18,10 @@
         top: 20px;
         right: 20px;
         cursor: pointer;
+
+        &:hover .svg-icon {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-summary/_deployment-summary.scss]: Can you also add
```
&:focus,
&:active,
```

File: jujugui/static/gui/src/app/components/machine-view/machine/_machine.scss [hunk 3]
```diff
@@ -67,6 +67,14 @@
         }
     }
 
+    &--machine, &--container {
+        transition: background-color 0.3s;
+
+        &:hover {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/machine-view/machine/_machine.scss]: Can you also add
```
&:focus,
&:active,
```

File: jujugui/static/gui/src/app/components/machine-view/machine/_machine.scss [hunk 4]
```diff
@@ -67,6 +67,14 @@
         }
     }
 
+    &--machine, &--container {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/machine-view/machine/_machine.scss]: Would you mind moving each of these class names to new lines?

**Comment by huwshimi** [comment]
Nice work @kadams54 QA OK :+1: with a few tiny fixes.

**Comment by kadams54** [comment]
:shipit: Thanks all!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4218/
Test PASSed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4219/
Test FAILed.

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1448

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**